lib/session.js

```diff
@@ -115,10 +133,25 @@ exports.signin = function(lookup, req, res, onSuccess, onFail) {
 
 exports.signout = function(req, res, next) {
 
-	res.clearCookie('keystone.uid');
-	req.user = null;
+	keystone.callHook(req.user, 'pre:signout', function(err) {
+		if (err) {
+			console.log("An error occurred in signout 'pre' middleware", err);
```

Comment:
It didn't seem like a good idea to potentially block a user from signing out just because some 'pre:signout' middleware throws. Safety seemed more important than allowing blocking logic. Can't really come up with a use case where you'd _really_ want to prevent a user from signing out TBH.
Obviously this could be solved by next-ticking the real signout call and throwing the error, sth like:

```js
res.clearCookie('keystone.uid');
req.user = null;

process.nextTick(function(){
	req.session.regenerate(function(err) {
		if (err) {
			return next(err);
		}
		keystone.callHook({}, 'post:signout', function(err) {
			if (err) {
				console.log("An error occurred in signout 'post' middleware", err);
			}
			next();
		});
	});
})

if (err) {
	throw err;
}
```